maintenance_request_purchase/tests/test_maintenance_purchase.py

```diff
@@ -36,3 +36,44 @@ def test_maintenance_purchase(self):
         requests = self.env[action["res_model"]].search(action["domain"])
         self.assertIn(self.request_1, requests)
         self.assertIn(self.request_2, requests)
+
+    def test_compute_total_purchase_amount(self):
+        # product 4 -> 500.0
+        self.po_1.write(
+            {
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.env.ref("product.product_product_4").id,
+                            "product_qty": 10,
+                        },
+                    )
+                ]
+            }
+        )
+        self.request_1.purchase_order_ids = self.po_1
+        self.assertEqual(self.request_1.total_purchase_amount, 5000)
+        self.po_2 = self.env["purchase.order"].create(
+            {
+                "partner_id": self.supplier.id,
+                "date_planned": "2017-02-11 22:00:00",
+            }
+        )
+        self.po_2.write(
+            {
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.env.ref("product.product_product_4").id,
+                            "product_qty": 5,
+                        },
+                    )
+                ]
+            }
+        )
+        self.request_1.purchase_order_ids |= self.po_2
+        self.assertEqual(self.request_1.total_purchase_amount, 7500)
```

Comment:
Data for this test is weak, because product data comes from demo products, that could be removed from a version to another.

Anyway, if you use the demo product, at this line you should retrieve product price for the assert, not assuming `500.0`, that it could be changed. Or even better, creating PO lines with a price, so you won't depend on demo product price.